src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContactBubbleStab.cpp

```diff
@@ -0,0 +1,1180 @@
+#include "physicsSolvers/contact/LogLevelsInfo.hpp"
+
+#include "constitutive/ConstitutiveManager.hpp"
+#include "constitutive/contact/FrictionSelector.hpp"
+#include "fieldSpecification/FieldSpecificationManager.hpp"
+
+
+namespace geos
+{
+
+using namespace constitutive;
+using namespace dataRepository;
+using namespace fields;
+
+SolidMechanicsLagrangeContactBubbleStab::SolidMechanicsLagrangeContactBubbleStab( const string & name,
+                                                                                  Group * const parent ):
+  ContactSolverBase( name, parent )
+{
+  m_faceTypeToFiniteElements["Quadrilateral"] =  std::make_unique< finiteElement::H1_QuadrilateralFace_Lagrange1_GaussLegendre2 >();
+  m_faceTypeToFiniteElements["Triangle"] =  std::make_unique< finiteElement::H1_TriangleFace_Lagrange1_Gauss1 >();
+
+}
+
+SolidMechanicsLagrangeContactBubbleStab::~SolidMechanicsLagrangeContactBubbleStab()
+{
+  // TODO Auto-generated destructor stub
+}
+
+real64 SolidMechanicsLagrangeContactBubbleStab::solverStep( real64 const & time_n,
+                                                            real64 const & dt,
+                                                            const integer cycleNumber,
+                                                            DomainPartition & domain )
+{
+  if( cycleNumber == 0 )
+  {
+    /// Apply initial conditions to the Fault
+    FieldSpecificationManager & fieldSpecificationManager = FieldSpecificationManager::getInstance();
+
+    forDiscretizationOnMeshTargets ( domain.getMeshBodies(), [&]( string const &,
+                                                                  MeshLevel & mesh,
+                                                                  arrayView1d< string const > const & )
+
+    {
+      fieldSpecificationManager.applyInitialConditions( mesh );
+      // Would like to do it like this but it is not working. There is a cast in Object path that tries to cast
+      // all objects that derive to ElementSubRegionBase to the specified type so this obviously faisl.
+      //   fieldSpecificationManager.forSubGroups< FieldSpecificationBase >( [&] ( FieldSpecificationBase const & fs )
+      //   {
+      //     if( fs.initialCondition() )
+      //     {
+      //       fs.apply< SurfaceElementSubRegion >( mesh,
+      //                                            [&]( FieldSpecificationBase const & bc,
+      //                                                 string const &,
+      //                                                 SortedArrayView< localIndex const > const & targetSet,
+      //                                                 SurfaceElementSubRegion & targetGroup,
+      //                                                 string const fieldName )
+      //       {
+      //         bc.applyFieldValue< FieldSpecificationEqual >( targetSet, 0.0, targetGroup, fieldName );
+      //       } );
+      //     }
+      //   } );
```

Comment:
I don't understand the explanation here.

Reply:
the way we loop over meshObjects is a bit different from some of the other loops.

Reply:
[Here](https://github.com/GEOS-DEV/GEOS/blob/d9bdfe715b6eb89674298a50bc43d8b1a3021d86/src/coreComponents/mesh/MeshObjectPath.hpp#L316) it looks like that we check if the object is a ElementSubRegionBase but not whether the type matches the `OBJECT_TYPE` specified. I could add it but:
1. I wonder if it's a design choice so I want to do it in a separate PR.
2. I think that if we do what we discussed to specify what field specifications are applied, all these small issues should go away.

Reply:
got it. I think that part of code is unfinished.